examples/snap-glib.c

```diff
@@ -121,6 +121,22 @@ static int info (int argc, char **argv)
 
         const gchar *publisher = snapd_snap_get_publisher_display_name (snap);
         const gchar *license = snapd_snap_get_license (snap);
+        g_autoptr(GError) interfaces_error = NULL;
+        g_autoptr(GPtrArray) interfaces = NULL;
+
+        interfaces = snapd_client_get_interfaces2_sync(client, SNAPD_GET_INTERFACES_FLAGS_NONE, NULL, NULL, &interfaces_error);
+
+        for (int i = 0; i < interfaces->len; ++i) {
+            SnapdInterface *interface = g_ptr_array_index(interfaces, i);
+            g_autofree const gchar *interface_name = snapd_interface_get_name(interface);
```

Comment:
Remove the g_autofree - you are freeing the name string that is owned by the SnapdInteraface (probably the cause of your segfault).